cmd/strelaysrv/listener.go

```diff
@@ -119,6 +119,15 @@ func protocolConnectionHandler(tcpConn net.Conn, config *tls.Config) {
 
 			switch msg := message.(type) {
 			case protocol.JoinRelayRequest:
+				if len(token) > 0 && msg.Token != token {
```

Comment:
`token` should be a parameter to this function, not accessing a global variable. The `len(token) > 0` check is typically spelled `token != ""` for strings.